esp-hal/src/spi/master.rs

```diff
@@ -884,7 +889,9 @@ where
     /// `with_sio0`.
     pub fn with_mosi(mut self, mosi: impl PeripheralOutput<'d>) -> Self {
         let mosi = mosi.into();
-        mosi.set_output_enable(false);
+
+        mosi.apply_output_config(&OutputConfig::default());
+        mosi.set_output_enable(true); // TODO turn this bool into a Yes/No/PeripheralControl trio
```

Comment:
Just checking that leaving this comment here was intentional.

Reply:
Yes, it's out of scope I think. Would make some things clearer, but it doesn't belong next to the other 1200 lines of change